src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/CourseActivity.kt

```diff
@@ -0,0 +1,175 @@
+        "incentiveLevel": "Basic",
+        "payBand": "A",
+        "rate": 150,
+        "pieceRate": 150,
+        "pieceRateItems": 10
+      }
+    ],
+    "riskLevel": "High", ----
+    "minimumIncentiveLevel": "Basic",
+    "startDate": "2022-12-23",
+    "endDate": "2022-12-23"
+  }
+     */
+  @Id
+  @SequenceGenerator(name = "CRS_ACTY_ID", sequenceName = "CRS_ACTY_ID", allocationSize = 1)
+  @GeneratedValue(generator = "CRS_ACTY_ID")
+  @Column(name = "CRS_ACTY_ID", nullable = false)
+  val courseActivityId: Long = 0,
+
+  @Column
+  @Size(max = 20)
+  var code: String? = null,
+
+  @Column
+  @Size(max = 6)
+  var caseloadId: String? = null,
+
+  @ManyToOne(optional = false, fetch = FetchType.LAZY)
+  @JoinColumn(name = "AGY_LOC_ID", nullable = false)
+  val prison: AgencyLocation,
+
+  @Column
+  @Size(max = 40)
+  var description: String? = null,
+
+  @Column
+  var capacity: Int? = null,
+
+  @Column(name = "ACTIVE_FLAG")
+  @Type(type = "yes_no")
+  val active: Boolean = false,
+
+  @ManyToOne(optional = false)
+  @JoinColumn(name = "PROGRAM_ID", nullable = false)
+  var program: ProgramService? = null,
+
+  @Column
+  val scheduleStartDate: LocalDate? = null,
+
+  @Column
+  val scheduleEndDate: LocalDate? = null,
+
+  @OneToMany(mappedBy = "courseActivity", cascade = [CascadeType.ALL], fetch = FetchType.LAZY)
+  var payRates: List<CourseActivityPayRate>? = null,
+
+  @Column
+  @Size(max = 12)
+  val caseloadType: String? = "INST",
+
+  @Column
+  @Size(max = 12)
+  val courseClass: String? = "COURSE",
+
+  @Column
+  @Size(max = 12)
+  val providerPartyClass: String? = "AGY",
+
+  @Column
+  @Size(max = 6)
+  val providerPartyCode: String? = prison.id,
+
+  @Column
+  @Size(max = 12)
+  val courseActivityType: String? = "PA",
+
+  @ManyToOne(fetch = FetchType.LAZY)
+  @JoinColumnsOrFormulas(
+    value = [
+      JoinColumnOrFormula(
+        formula = JoinFormula(
+          value = "'" + IEPLevel.IEP_LEVEL + "'",
+          referencedColumnName = "domain"
+        )
+      ),
+      JoinColumnOrFormula(column = JoinColumn(name = "IEP_LEVEL", referencedColumnName = "code"))
+    ]
+  )
+  val iepLevel: IEPLevel,
+
+  @ManyToOne(optional = false)
+  @JoinColumn(name = "INTERNAL_LOCATION_ID", nullable = false)
+  var internalLocation: AgencyInternalLocation? = null,
+
+  @Column(name = "HOLIDAY_FLAG")
+  @Type(type = "yes_no")
+  val holiday: Boolean = false, // If the course/activity conforms to national holidays
+
+  @Column
+  var payPerSession: PayPerSession? = null,
+) {
+  override fun equals(other: Any?): Boolean {
+    if (this === other) return true
+    if (other == null || Hibernate.getClass(this) != Hibernate.getClass(other)) return false
+    other as CourseActivity
+
+    return courseActivityId == other.courseActivityId
+  }
+
+  override fun hashCode(): Int = javaClass.hashCode()
+
+  @Override
+  override fun toString(): String {
+    return this::class.simpleName + "(courseActivityId = $courseActivityId )"
```

Comment:
maybe add description? might be useful when debugging?

Reply:
👍